src/modules/modem_module.c

```diff
@@ -254,43 +254,94 @@ static int modem_data_get(void)
 	struct modem_module_event *modem_module_event =
 			new_modem_module_event();
 
-	modem_module_event->data.modem.rsrp = rsrp_value_latest;
-	modem_module_event->data.modem.ip =
-		modem_param.network.ip_address.value_string;
-	modem_module_event->data.modem.cell = modem_param.network.cellid_dec;
-	modem_module_event->data.modem.mccmnc =
-		modem_param.network.current_operator.value_string;
-	modem_module_event->data.modem.area =
-			modem_param.network.area_code.value;
-	modem_module_event->data.modem.appv = CONFIG_CAT_TRACKER_APP_VERSION;
-	modem_module_event->data.modem.brdv = modem_param.device.board;
-	modem_module_event->data.modem.fw =
-		modem_param.device.modem_fw.value_string;
-	modem_module_event->data.modem.iccid =
+	modem_module_event->data.modem_static.app_version =
+			CONFIG_CAT_TRACKER_APP_VERSION;
+
+	modem_module_event->data.modem_static.board_version =
+			modem_param.device.board;
+
+	modem_module_event->data.modem_static.modem_fw =
+			modem_param.device.modem_fw.value_string;
+
+	modem_module_event->data.modem_static.iccid =
 			modem_param.sim.iccid.value_string;
-	modem_module_event->data.modem.nw_lte_m =
-		modem_param.network.lte_mode.value;
-	modem_module_event->data.modem.nw_nb_iot =
-		modem_param.network.nbiot_mode.value;
-	modem_module_event->data.modem.nw_gps =
+
+	modem_module_event->data.modem_static.nw_mode_ltem =
+			modem_param.network.lte_mode.value;
+
+	modem_module_event->data.modem_static.nw_mode_nbiot =
+			modem_param.network.nbiot_mode.value;
+
+	modem_module_event->data.modem_static.nw_mode_gps =
 			modem_param.network.gps_mode.value;
-	modem_module_event->data.modem.bnd =
-		modem_param.network.current_band.value;
-	modem_module_event->data.modem.mod_ts = k_uptime_get();
-	modem_module_event->data.modem.mod_ts_static = k_uptime_get();
-	modem_module_event->data.modem.queued = true;
-	modem_module_event->type = MODEM_EVT_MODEM_DATA_READY;
+
+	modem_module_event->data.modem_static.band =
+			modem_param.network.current_band.value;
+
+	modem_module_event->data.modem_static.timestamp = k_uptime_get();
+
+	modem_module_event->type = MODEM_EVT_MODEM_STATIC_DATA_READY;
 
 	EVENT_SUBMIT(modem_module_event);
 
 	return 0;
 }
 
-static bool modem_data_requested(enum app_module_data_type *data_list,
-				size_t count)
+static int dynamic_modem_data_get(void)
+{
+	int err;
+
+	/* Request data from modem information module. */
+	err = modem_info_params_get(&modem_param);
+	if (err) {
+		LOG_ERR("modem_info_params_get, error: %d", err);
+		return err;
+	}
+
+	struct modem_module_event *modem_module_event =
+			new_modem_module_event();
+
+	modem_module_event->data.modem_dynamic.rsrp =
+			rsrp_value_latest;
+
```

Comment:
The blank lines between variable assignments are not really needed

Reply:
woops. I merged before I saw your comment.